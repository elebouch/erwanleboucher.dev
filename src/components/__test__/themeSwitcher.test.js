import React from "react"
import { render, fireEvent } from "@testing-library/react"
import "jest-styled-components"

import ThemeSwitcher from "../themeSwitcher"

describe("ThemeSwitcher", () => {
  it("renders correctly", () => {
    const { container } = render(<ThemeSwitcher />)
    expect(container.innerHTML).toMatchSnapshot()
  })

  it("it should be checked when pressed and theme has been changed", () => {
    const { getByTestId } = render(<ThemeSwitcher />)
    const checkbox = getByTestId("theme-checkbox")
    expect(checkbox.checked).toEqual(false)
    fireEvent.click(checkbox)
    expect(checkbox.checked).toEqual(true)
    expect(document.documentElement.getAttribute("data-theme")).toEqual("dark")
  })
})
