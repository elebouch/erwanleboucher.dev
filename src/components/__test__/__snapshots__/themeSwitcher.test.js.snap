// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemeSwitcher renders correctly 1`] = `"<input type=\\"checkbox\\" id=\\"switch\\" data-testid=\\"theme-checkbox\\" class=\\"sc-AxirZ cymzFC\\"><label for=\\"switch\\" class=\\"sc-AxjAm rpoTh\\"></label>"`;
